packages/cli/internal/pkg/cli/logs_workflow.go

```diff
@@ -72,11 +82,40 @@ func (o *logsWorkflowOpts) Execute() error {
 		return err
 	}
 	log.Debug().Msgf("Showing logs for workflow run '%s'", o.runId)
-	jobIds, err := o.getJobIds()
+
+	runLog, err := o.workflowManager.GetRunLog(o.runId)
 	if err != nil {
 		return err
 	}
 
+	var jobIds []string
+	if o.taskId != "" {
+		var taskExists = false
+		for _, task := range runLog.Tasks {
+			if task.JobId == o.taskId {
+				taskExists = true
+				break
+			}
+		}
+
+		if !taskExists {
+			log.Info().Msgf("Task `%s` does not exist for run `%s`", o.taskId, o.runId)
+			return nil
+		}
+
+		jobIds = []string{o.taskId}
+	} else if o.allTasks || o.failedTasks {
+		jobIds, err = o.getJobIds(runLog.Tasks)
+		if err != nil {
+			return err
+		}
+	} else {
+		b := bytes.NewBufferString("")
+		format.NewStringFormatter(b).Write(runLog)
+		printLn(b)
```

Comment:
Could you please clarify what this code does? Can it be simplified?

Reply:
It's getting a formatted version of the RunLog struct and printing it out with the existing log print logic. If we don't use `printLn` to feed stdout for logs, it would require additional changes to our unit testing. Since that's how we validate log output

Reply:
My concern here is that "formatted RunLog struct" is not a log, so "log printing logic" is not applicable to it. I think it should be printed in a similar to all other command results way.